### ISSUE TITLE
Clean-up of configuration options

### ISSUE THREAD

**Issue by pdelboca** [ckan/ckan#7671]
### Intro

When working on #7662 I ran into some configs like: `search.facets.limit` or `search.facets.default` that didn't make a lot of sense to me. What's the difference between having 5, 7, 8? and What's the value of having a whole config for something that can be easily override in a template.

After that, while reading a book I found this chapter about [Avoid Preferences](https://basecamp.com/gettingreal/06.4-avoid-preferences) that really resonated with my experience while working with CKAN.

Taking into consideration that there's a difference between building a product and a framework I do think that we are over-using preferences to avoid difficult decision. 

### Point

I think CKAN would greatly benefit from lesser configuration options, both in terms of complexity and maintainability. Even more in the area of the front-end where we have preferences whose effect can be also achieved by overriding the template (a common practice in CKAN) 

### Proposal

Let's get rid of some of our configurations and lets make CKAN more lean in terms of it's codebase. Let's discuss which configuration options make us happy and make CKAN a good framework and which ones just doesn't make any sense.

### Proposed configurations to remove:
 - [ ] `search.facets.limit` and `search.facets.default`
 - [ ] `ckan.datasets_per_page`
 - [ ] `ckan.dumps_url`
 - [ ] `ckan.dumps_format`
 - [ ] `ckan.featured_groups`
 - [ ] `ckan.featured_orgs`
 - [ ] `ckan.default_group_sort`
 - [ ] `ckan.homepage_style` (why do we have 3 options?)
 - [ ] `ckan.template_head_end`
 - [ ] `ckan.template_footer_end`
 - [ ] `ckan.template_title_delimiter` What's the goal of this one?
 - [ ] `ckan.admin_tabs`
 - [ ] `ckan.user_list_limit`
 - [ ] `ckan.activity_list_limit`
 - [ ] ckan.activity_list_max`
 - [ ] ...
 - [ ] ...

Which preferences would you think it makes sense to remove?

**Comment by wardi** [comment]
we could render template titles with a helper the same way we do many other titles, that would let us override this delimiter, provide smart translations and more

**Comment by pdelboca** [comment]
@Pradyumn-NEC are you interested in working on some of this configs?

`ckan.default_group_sort` should be quite straight forward to remove. Do no forget to add a CHANGELOG file mentioning that this config has been removed will no longer be available.

**Comment by Pradyumn-NEC** [comment]
@pdelboca Yes sure. I am interested in working on this issue. I will raise PR for ```ckan.default_group_sort```.

**Comment by Vishal-NEC** [comment]
Hi @pdelboca I Am Interested in working on this issue.

**Comment by Vishal-NEC** [comment]
@pdelboca Can you please confirm that only we need to remove `ckan.default_group_sort` or need to remove above mentioned configurations like `search.facets.limit` , `search.facets.default` ,`ckan.featured_groups` , `ckan.featured_orgs` and others?

**Comment by pdelboca** [comment]
@Vishal-NEC I confirm. You can start by `ckan.featured_groups` and `ckan.featured_orgs`.

Please, do not forgive to add a change file pointing that the configuration is removed. Also, note that this two configurations are used to render things in the homepage, so all relevant code in templates needs to be removed as well.

**Comment by Gauravp-NEC** [comment]
@pdelboca , I will submit a PR to remove  `ckan.featured_orgs` configuration.

**Comment by Vishal-NEC** [comment]
Hi @pdelboca I have Raise a PR for removing `ckan.featured_group` Please Review it and let me know if any other changes required.

**Comment by Vishal-NEC** [comment]
Hi @pdelboca Please review the PR #7929 for removing config `ckan.featured_groups`. Please let me know if any other changes required.

**Comment by Gauravp-NEC** [comment]
@pdelboca, As this issue is still open. Please let me know what is left to do in this issue.

**Comment by Labuschagne-Miro** [comment]
@pdelboca , I would like to take all the remaining task.
